core/parse-xml.c

```diff
@@ -535,6 +535,11 @@ static void get_uint16(char *buffer, uint16_t *i)
 	*i = atoi(buffer);
 }
 
+static void get_uint32(char *buffer, int32_t *i)
```

Comment:
What is your reason to use uint32 and not good old int as we do everywhere else?

Reply:
That's apparently a typo, as the output parameter is `int32_t`, not `uint32_t`.

Reply:
Typo

Reply:
Fine, but why a fixed precision type?

Reply:
Because that reflects the `pressure_t` type. That makes sense to me.

Reply:
Could you please make sure dive planner.cpp also uses your new helper functions? 

The other thing is (I said this before and @bstoeger did not really agree so please make of it what you like): I find using fixed precision types like int32_t strange unless you are doing some bit fiddling where you need control over how many bits you really have. Why not use int or our own semantically dimensional types like pressure_t? And yes, there might be good reasons to use unsigned types (like: pressure cannot be negative) but using those has bitten us already in other places (as unsigned int u = 0; int i = -1; if (u < i) printf("WTF!\n") does print since i is cast to unsigned int (and thus is very big) rather than casting u to int.

Reply:
I am trying to limit the code changes to deal with atmospheric pressure in the UI and it is much more invasive than I expected. Would it be ok if I dealt with that as a different PR? Once this one is done, I will do the changes to the planner immediately. Ok?

The mbar member of pressure_t is int32_t, that was the logic. But the airpressure is now being parsed using the pressure() function that operates on a pressure_t structure.

Reply:
With "fixed" I meant defined as in "platform independent". With int you get what is natural for the platform (or the compiler more specifically), the "natural" thing whereas specifying the number of bits forces that on the compiler. But this really not worth losing sleep about. Just don't forget to adopt dive planner.cpp to use the helper as well soon.

Reply:
> With "fixed" I meant defined as in "platform independent". 

Yes, I got that now, but with your comment 9 days ago I was a bit confused, as I thought you were arguing against integer-arithmetic... I think the proper name for the int*_t types is "exact width" or "fixed width" as opposed to "fixed precision", which for me means a fixed number of decimal places.

> But this really not worth losing sleep about.

True. 😁